### PR TITLE
✨ Add nft subscription minting

### PR DESCRIPTION
**Pull request by williamchong**
For first release, no link to subscription page would be presented for common users
It is expected we would distribute such link by invite or via other channel
Subscription related status would only appear when user is a subscriber

### CODE REVIEW

File: pages/nft/iscn/_iscnId.vue [hunk 1]
```diff
@@ -798,39 +845,52 @@ export default class NFTTestMintPage extends Vue {
 
   async createNftClass() {
     await this.initIfNecessary()
-    if (!this.signer) return
+    if (!this.signer) return '';
```
Comment (nwingt) [inline pages/nft/iscn/_iscnId.vue:848]: Should this guard be put in the else below like L960?

**Review by nwingt — APPROVED** [review]

**Review by WeiJun0827 — APPROVED** [review]
Was thinking about whether we should handle the new subscription logic in `_iscnId.vue` with the strategy pattern. But maybe let it be for now?

**Comment by williamchong** [comment]
> Was thinking about whether we should handle the new subscription logic in `_iscnId.vue` with the strategy pattern. But maybe let it be for now?

Seems to be a good idea, not sure how to implement in nuxt context through.
Do we need to pass `this` into constrcutor if we always call `this` in methods?

File: pages/nft/iscn/_iscnId.vue (hunk 1 above)
Comment (williamchong) [reply pages/nft/iscn/_iscnId.vue:848]: yes indeed

**Comment by WeiJun0827** [comment]
> Seems to be a good idea, not sure how to implement in nuxt context through. Do we need to pass `this` into constrcutor if we always call `this` in methods?

Yeah, seems can do.

**Pull request merged**